lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorRuntime.cs

```diff
@@ -254,21 +253,12 @@ private void OnException(Exception e)
 
                 try
                 {
-                    using (var memStream = new MemoryStream())
-                    {
-                        var formatter = new BinaryFormatter();
-                        formatter.Serialize(memStream, e);
-                        errorBytes = memStream.ToArray();
-                    }
+                    errorBytes = ByteUtilities.SerializeToBinaryFormat(e);
                 }
                 catch (SerializationException se)
                 {
-                    using (var memStream = new MemoryStream())
-                    {
-                        var formatter = new BinaryFormatter();
-                        formatter.Serialize(memStream, new NonSerializableEvaluatorException(e.ToString(), se));
-                        errorBytes = memStream.ToArray();
-                    }
+                    errorBytes = ByteUtilities.SerializeToBinaryFormat(
+                        new NonSerializableEvaluatorException(e.ToString(), se));
```

Comment:
In HandleTaskException, if the exception is not serializable, your wrap it into a serializable exception. Here you create a NonSerializableEvaluatorException, why use different approaches?